### ISSUE TITLE
How do I make DAGs become known to the web-server?

### ISSUE THREAD

**Issue by davidaventimiglia-personal** [apache/airflow#116]
Hi,

How do I make DAGs become known to the web-server?  If I create a Python-scripted dag file, place it into the dags_folder, and then restart the web-server, the new DAG is visible under the first "DAGs" screen (though not under the second screen, available from the "Admin" drop-down menu).  That's great.  But then if I create another new DAG file and place it into the dags_folder, it is not known to the web-server until I restart it.  I thought that there might be a CLI choice for adding a DAG, but I found none.  Thanks.

**Comment by mistercrunch** [comment]
Yes, this is confusing in the current state. If you happen to be running an `airflow scheduler`, it will monitor your `DAGS_FOLDER` and keep the list up to date.

There's an hidden endpoint at `/admin/airflow/refresh_all` that will refresh the web server's DagBag from scratch.

Also note that if you run the web server in debug mode `airflow webserver --debug`, it will refresh your dags as you update them, though there are limitations here.

The challenge is that in our environment we have multiple multi-threaded server, and that each thread has its own DagBag. With many DAGs, some very large, refreshing the DagBags in the scope of a web request is a bad idea. Anyhow, we'll have 100% stateless web servers soon as we move to run everything off of serialized, versioned DAGs that live in the DB. This may be a 2.0 kinda thing.

**Comment by davidaventimiglia-personal** [comment]
Thanks!

**Comment by davidaventimiglia-personal** [comment]
Sadly, neither of those things seemed to work with the version of AirFlow that I have (installed via pip).  I started the web-server with the `--debug` flag.  That failed to pick up new DAGs that I dropped into the `DAGS_FOLDER`.  I also hit the endpoint `/admin/airflow/refresh_all` after which the display changed to include the status message "All DAGs are now up to date."  However, nothing else changed, and no DAG was picked up.  I then stopped and re-started the server, and the new DAG appeared.
